cmd/frontend/internal/app/ui/sveltekit/gen/gen_routes.go

```diff
@@ -0,0 +1,303 @@
+	})
+}
+
+func writeTSFile(dest string, routes []routeInfo) error {
+	t := template.Must(template.New("routes").Funcs(template.FuncMap{
+		"isRepoRoot": func(route routeInfo) bool {
+			for _, tag := range route.Tags {
+				if tag == "RepoRoot" {
+					return true
+				}
+			}
+			return false
+			},
+		}).Parse(tsFileTemplate))
+
+	// Write template to _routes.ts file
+	file, err := os.Create(dest)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+	return t.Execute(file, map[string]interface{}{
+		"Routes": routes,
+	})
+}
+
+func getPagePaths() []string {
+	var paths []string
+
+	// When running under Bazel, the page file paths are passed as arguments.
+	args := flag.Args()
+	if len(args) > 0 {
+		return args
+	}
+
+	err := filepath.Walk(SRC_ROUTES_PREFIX, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if filepath.Base(path) == "+page.svelte" {
+			paths = append(paths, path)
+		}
+		return nil
+	})
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error walking directory: %v", err)
+		os.Exit(1)
+	}
+	return paths
+}
+
+var (
+	splitPattern = regexp.MustCompile(`\s+`)
+	tagPattern = regexp.MustCompile(`^\s*//\s+@sg\s+`)
+	groupPattern = regexp.MustCompile(`^\([^)]+\)$`)
+	paramPattern = regexp.MustCompile(`^(\[)?(\.\.\.)?(\w+)(?:=(\w+))?(\])?$`)
+	restParamPattern = regexp.MustCompile(`^\[\.\.\.(\w+)(?:=(\w+))?\]$`)
+	optionalParamPattern = regexp.MustCompile(`^\[\[(\w+)(?:=(\w+))?\]\]$`)
+)
+
+// extractTags finds lines of the forms
+//     // @sg tag1 tag2 tag3
+// and returns the tags found.
+func getRouteInfo(path string) (routeInfo, error) {
+	var info routeInfo
+
+	f, err := os.Open(path)
+	if err != nil {
+		return info, err
+	}
+	defer f.Close()
+
+	// All routes are opt-in by default
+	tagNames := []string{"EnableOptIn"}
+
+
+	scanner := bufio.NewScanner(f)
+	for scanner.Scan() {
+		line := scanner.Text()
+		if tagPattern.MatchString(line) {
+			match := tagPattern.FindString(line)
+			tagNames = append(tagNames, splitPattern.Split(strings.TrimPrefix(line, match), -1)...)
+			break
+		}
+	}
+
+	for _, tag := range tagNames {
+		if !tags.IsTagValid(tag) {
+			return info, errors.Newf("Invalid tag '%s'. Valid tags: %s", tag, tags.AvailableTags())
+		}
+	}
+	info.Tags = tagNames
+
+	if err := scanner.Err(); err != nil {
+		return info, err
+	}
+
+	return info, nil
+}
+
+// Map SvelteKit specific parameter matchers to regular expressions. This is a "best effort" approach
+// because parameter matchers in SvelteKit are functions that can perform arbitrary logic.
+var paramMatchers = map[string]string{
+	"reporev": "/" + routevar.RepoPatternNonCapturing + `(?:@` + routevar.RevPatternNonCapturing + `)?`,
+}
+
+// This code follows the regex generation logic from
+// https://github.com/sveltejs/kit/blob/main/packages/kit/src/utils/routing.js
+func patternForRouteId(routeId string) (string, error) {
+	if routeId == "/" {
+		return "^/$", nil
+	}
+
+	b := strings.Builder{}
```

Comment:
style nit: it's more idiomatic to use the zero-value

```suggestion
	var b strings.Builder
```